File: core/recc/packet/container.py
# -*- coding: utf-8 -*-

from typing import Optional, Dict, List
from dataclasses import dataclass
from enum import Enum


class ContainerOperator(Enum):

    Start = 0
    Stop = 1
    Kill = 2
    Restart = 3
    Pause = 4
    Resume = 5
    Remove = 6

    @classmethod
    def from_str(cls, text: str):
        operator_key = text[0].upper() + text[1:].lower()
        operators = [s for s in dir(cls) if not s.startswith("_")]
        if operator_key in operators:
            return getattr(cls, operator_key)
        raise KeyError(f"Not found '{operator_key}' enum in {cls.__name__}")


@dataclass
class ContainerA:
    key: str
    name: str
    status: str
    image: str
    created: str
    labels: Dict[str, str]
    ports: Dict[str, List[str]]


@dataclass
class ControlContainersQ:
    keys: List[str]
    operator: str
    signal: Optional[str] = None
    force: Optional[bool] = None

File: fe/src/packet/container.ts
export interface ContainerA {
    key: string;
    name: string;
    status: string;
    image: string;
    created: string;
    labels: object;
    ports: object;
}

export const STATUS_CREATED = 'Created';
export const STATUS_RESTARTING = 'Restarting';
export const STATUS_RUNNING = 'Running';
export const STATUS_REMOVING = 'Removing';
export const STATUS_PAUSED = 'Paused';
export const STATUS_EXITED = 'Exited';
export const STATUS_DEAD = 'Dead';

export interface ControlContainersQ {
    keys: Array<string>;
    operator: string;
    signal?: string;
    force?: boolean;
}

export const CONTROL_OPERATOR_START = 'Start';
export const CONTROL_OPERATOR_STOP = 'Stop';
export const CONTROL_OPERATOR_KILL = 'Kill';
export const CONTROL_OPERATOR_RESTART = 'Restart';
export const CONTROL_OPERATOR_PAUSE = 'Pause';
export const CONTROL_OPERATOR_RESUME = 'Resume';
export const CONTROL_OPERATOR_REMOVE = 'Remove';
